capsule_cabinet.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+# A script to generate cabinets for fwupd / LVFS
+
+set -e
+
+function die() {
+    echo error: "$@" 1>&2
+    exit 1
+}
+
+if [ ! -f .config ]; then
+    die "no '.config' file in current directory"
+fi
+
+# import coreboot's config file replacing $(...) with ${...}
+while read -r line; do
+    if ! eval "$line"; then
+        die "failed to source '.config'"
+    fi
+done <<< "$(sed 's/\$(\([^)]\+\))/${\1}/g' .config)"
+
+if [ "$CONFIG_DRIVERS_EFI_UPDATE_CAPSULES" != y ]; then
+    die "current board configuration lacks support of update capsules"
+fi
+
+if [ -z $1 ]; then
+    die "No input capsule specified"
+fi
+
+if [ ! -f $1 ]; then
+    die "File $1 not found"
+fi
+
+capsule=$1
+date=$(stat -c %w $capsule | cut -d ' ' -f 1)
+vendor=$(cat .config | grep -e "CONFIG_VENDOR_.*=y" | cut -d '=' -f 1 | cut -d '_' -f 3- | awk '{ print tolower($0) }')
+version=$(echo $CONFIG_LOCALVERSION | tr -d 'v' | cut -d '-' -f 1)
+
+archive_dir=$(mktemp --tmpdir -d XXXXXXXX)
+
+cat > "${archive_dir}/firmware.metainfo.xml" << EOF
+<?xml version='1.0' encoding='utf-8'?>
+<component type="firmware">
+  <id>com.${vendor}.${CONFIG_MAINBOARD_SMBIOS_PRODUCT_NAME}.${CONFIG_MAINBOARD_VERSION}.system.firmware</id>
+  <name>${CONFIG_MAINBOARD_SMBIOS_PRODUCT_NAME}</name>
+  <summary>${CONFIG_MAINBOARD_SMBIOS_PRODUCT_NAME} ${CONFIG_MAINBOARD_VERSION} system firmware</summary>
+  <description>
+    <p>Dasharo ${CONFIG_MAINBOARD_SMBIOS_PRODUCT_NAME} ${CONFIG_MAINBOARD_VERSION} system firmware</p>
+  </description>
+  <provides>
+    <firmware type="flashed">${CONFIG_DRIVERS_EFI_MAIN_FW_GUID}</firmware>
+  </provides>
+  <url type="homepage">https://docs.dasharo.com/</url>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>LicenseRef-proprietary</project_license>
+  <categories>
+    <category>X-System</category>
+  </categories>
+  <custom>
+    <value key="LVFS::VersionFormat">quad</value>
+    <value key="LVFS::VersionFormat">dell-bios-msb</value>
+    <value key="LVFS::UpdateProtocol">org.uefi.capsule</value>
+  </custom>
+  <releases>
+    <release version="${version}" date="${date}" tag="${CONFIG_LOCALVERSION}" urgency="high">
+      <checksum filename="firmware.bin" target="content"/>
```

Comment:
Since it's a capsule, maybe use `.cap` instead of `.bin`?

Reply:
fwupd requires that the firmware file inside the cabinet is named firmware.bin

Reply:
It seems to be the default name without the `<checksum>` tag: https://github.com/fwupd/fwupd/blob/15a1adfeb7c4010f38972fb0eb4ce2d2d040d9c6/src/fu-cabinet.c#L138-L141

A custom name is also used in the example metadata: https://lvfs.readthedocs.io/en/latest/metainfo.html

But we can keep it as is if there are compatibility concerns.